deepinv/tests/test_trainer.py

```diff
@@ -733,24 +733,36 @@ def forward(self, *args, **kwargs):
         )
 
 
-def test_total_loss(dummy_dataset, imsize, device, dummy_model, tmpdir):
+class ConstantLoss2(ConstantLoss):
+    pass
+
+
```

Comment:
I would like a trainer test using 4 dummy losses that cover the below possibilities:

1. Are evaluated differently at train vs test time
2. Adapt models that behave differently at train vs test time

The test should test that during various stages of training/val/eval, the losses are always evaluated correctly.